bitarray/_bitarray.c

```diff
@@ -2945,8 +2912,8 @@ static PyTypeObject Bitarraytype = {
     0,                                        /* tp_compare */
     (reprfunc) bitarray_repr,                 /* tp_repr */
     0,                                        /* tp_as_number*/
-    0,                                        /* tp_as_sequence */
-    0,                                        /* tp_as_mapping */
+    &bitarray_as_sequence,                    /* tp_as_sequence */
+    &bitarray_as_mapping,                     /* tp_as_mapping */
```

Comment:
So here you are making bitarray comply with the Mapping Protocol and the Sequence Protocol. And this is a lower hook into the interpreter than using th e__special__methods. And when calling len(), python will pick either the sequence length method or the mapping length method which resolve to the same thing. Not familiar with doing it directly in C, so I had to read through it. 

There is some information in the docs about the length method being preferred to calling len because of behavior on 32 bit systems...  any way we could fix that, or is that indicative of the types used in 32 bit python.

And is there any benefit in converting these functions into C code and getting rid of the py file? https://github.com/diamondman/bitarray/blob/master/bitarray/__init__.py#L80